src/LlbcAst.ml

```diff
@@ -1,10 +1,22 @@
-open Identifiers
 open Names
 open Types
 open Values
 open Expressions
+open Identifiers
 
 module FunDeclId = IdGen ()
+module GlobalDeclId = IdGen ()
+
+(* Strict type for the number of function declarations (see [global_to_fun_id] below) *)
+type global_id_converter = { fun_count : int }
+[@@deriving show]
+
+(** Converts a global id to its corresponding function id.
+    To do so, it adds the global id to the number of function declarations :
+    We have the bijection `global_id <=> fun_id + fun_id_count`.
+*)
+let global_to_fun_id (conv : global_id_converter) (gid : GlobalDeclId.id) : FunDeclId.id =
+  FunDeclId.of_int ((GlobalDeclId.to_int gid) + conv.fun_count)
```

Comment:
For now I don't need to have it outside of `LlbcOfJson`. If an access to a global is still needed, I think I'd prefer to put there a global id instead of computing it (e.g. in `fun_decl` with a global body).

Reply:
I saw the changes, and I agree with them :)